### PR TITLE
Connected to database

### PR DESCRIPTION
**Pull request by MandemGibson**

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  .then(() => console.log("Connected to database"))
+  .catch((err) => console.error("Error connecting to database: ", err.stack));
+
+async function closeConnection() {
+  client
+    .end()
+    .then(() => console.log("Connection ended"))
+    .catch((err) => console.error("Error closing connection: ", err.stack));
+}
+
```
Comment (bra-alex) [inline db.js:23]: Don't use promises in async functions. Use async await and try catch for the errors

File: db.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  try {
+    res = await client.query("SELECT * FROM todo");
+    if (res.rowCount === 0) {
+      console.log("No tasks found");
+      return;
+    } else {
+      res.rows.forEach((row) =>
+        console.log(`${row.id}. ${row.task} [${row.done ? "x" : ""}]`)
+      );
+    }
```
Comment (bra-alex) [inline db.js:34]: if you're using a return, there is no need for you to add an else block

File: db.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+async function markTaskAsDone(id) {
+  try {
+    res = await client.query(
+      "UPDATE todo SET done = true WHERE id = $1 RETURNING *",
+      [id]
+    );
+    await client.query("UPDATE todo SET updatedAt = CURRENT_TIMESTAMP WHERE id = $1 RETURNING *", [id])
+    console.log(`Marked "${res.rows[0].task}" as done`);
```
Comment (bra-alex) [inline db.js:62]: Can't all this be one query or a transaction?

File: db.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+async function markTaskANotsDone(id) {
+  try {
+    res = await client.query(
+      "UPDATE todo SET done = false WHERE id = $1 RETURNING *",
+      [id]
+    );
+    await client.query("UPDATE todo SET updatedAt = CURRENT_TIMESTAMP WHERE id = $1 RETURNING *", [id])
+    console.log(`Marked "${res.rows[0].task}" as not done`);
```
Comment (bra-alex) [inline db.js:77]: Can't all this be one query or a transaction?

File: db.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+async function fetchTodos() {
+  try {
+    res = await client.query("SELECT * FROM todo");
+    if (res.rowCount === 0) {
+      console.log("No tasks found");
+      return;
+    } else {
+      res.rows.forEach((row) =>
+        console.log(`${row.id}. ${row.task} [${row.done ? "x" : ""}]`)
```
Comment (bra-alex) [inline db.js:32]: You can map the response and transform it into the format you want. So you can do something like 
```javascript
res.rows.map(row => `${row.id}. ${row.task} [${row.done ? "x" : ""}]`)
```

File: index.js [hunk 6]
```diff
@@ -22,63 +31,54 @@ program
...
-    const todos = readToDos();
+    const todos = removeTodo(index);
     if (index > todos.length || index < 1) {
       console.log("Invalid task number");
       return;
     }
-    todos.splice(index - 1, 1);
-    writeToDos(todos);
-    console.log(`Task ${index} has been removed`);
   });
```
Comment (bra-alex) [inline index.js:75]: Isn't this check supposed to be before the marking?

**Review by bra-alex — CHANGES_REQUESTED** [review]

File: db.js (hunk 2 above)
Comment (MandemGibson) [reply db.js:34]: yessir

File: db.js (hunk 3 above)
Comment (MandemGibson) [reply db.js:62]: Yeah i think that would be better. Thanks

File: db.js (hunk 5 above)
Comment (MandemGibson) [reply db.js:32]: It's a cli project i don't know if it would be shown on the terminal without the console log but i go try

File: index.js (hunk 6 above)
Comment (MandemGibson) [reply index.js:75]: I added that for when someone mistakenly marks a task as done and want to undo it.

File: db.js (hunk 5 above)
Comment (bra-alex) [reply db.js:32]: you can do that, assign it to a variable and then print it out

Comment (MandemGibson) [reply db.js:32]: @bra-alex without forEach it would print out an array containing the tasks instead of printing the tasks line by line

**Pull request merged**